hrms/hr/doctype/leave_block_list/leave_block_list.py

```diff
@@ -49,30 +68,30 @@ def get_applicable_block_lists(employee=None, company=None, all_lists=False):
 		company = frappe.db.get_value("Employee", employee, "company")
 
 	def add_block_list(block_list):
-		if block_list:
-			if all_lists or not is_user_in_allow_list(block_list):
-				block_lists.append(block_list)
+		for d in block_list:
+			if all_lists or not is_user_in_allow_list(d):
+				block_lists.append(d)
 
 	# per department
 	department = frappe.db.get_value("Employee", employee, "department")
 	if department:
 		block_list = frappe.db.get_value("Department", department, "leave_block_list")
-		add_block_list(block_list)
+		block_list_leave_type = frappe.db.get_value("Leave Block List", block_list, "leave_type")
+		if not block_list_leave_type or not leave_type or block_list_leave_type == leave_type:
+			add_block_list([block_list])
 
 	# global
-	for block_list in frappe.db.sql_list(
-		"""select name from `tabLeave Block List`
-		where applies_to_all_departments=1 and company=%s""",
-		company,
-	):
-		add_block_list(block_list)
+	conditions = {"applies_to_all_departments": 1, "company": company}
+	if leave_type:
+		conditions["leave_type"] = leave_type
```

Comment:
You will have to add `or_filters` here to check:
- if the leave type matches
- or the leave type is not set in the list (meaning it's applicable for all leave types)

Currently, it does not work in the second case where the block list is not set up leave-type-wise.

Reply:
Thank you, have fixed the same :)